app/models/stacks_media_stream.rb

```diff
@@ -0,0 +1,69 @@
+##
+# media stream via stacks
+class StacksMediaStream < StacksFile
+  TYPE_TO_MANIFEST_FILE = {
+    hls: 'playlist.m3u8',
+    dash: 'manifest.mpd'
+  }.freeze
+
+  attr_accessor :format
+
+  def to_playlist_url
+    streaming_url_for(:hls)
+  end
+
+  def to_manifest_url
+    streaming_url_for(:dash)
+  end
+
+  private
+
+  def video?
+    Settings.stream.video.include? file_extension
+  end
+
+  def audio?
+    Settings.stream.audio.include? file_extension
+  end
+
+  def streaming_file_prefix
+    if video?
+      'mp4'
+    elsif audio?
+      'mp3'
+    end
+  end
+
+  def streaming_url_for(type)
```

Comment:
Consider placing this method at the top of the private section, since it is the main private interface of the public interface.